src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -4486,7 +4466,16 @@ def _contains_intrinsic_reduction(pnodes):
                                           ("dim", Literal(str(idx),
                                                           integer_type))]))
             else:
-                loop.addchild(mask_shape[idx-1].upper.copy())
+                lbound = mask_shape[idx-1].lower
+                if isinstance(lbound, Literal) and lbound.value == "1":
+                    # Lower bound is unity so size is just the upper bound.
+                    expr2 = mask_shape[idx-1].upper.copy()
+                else:
+                    # Size = upper-bound - lower-bound + 1
+                    expr = BinaryOperation.create(
+                        sub_op, mask_shape[idx-1].upper.copy(), lbound.copy())
+                    expr2 = BinaryOperation.create(add_op, expr, one.copy())
+                loop.addchild(expr2)
```

Comment:
Nice improvement but I am unsure if this covers the case where there is a non-unity lower but no upper, like when remapping a argument lowerbound:
```
subroutine test(a)
   integer, dimension(5:) :: a
```
Could you add a test?

Reply:
I've done my best. If it's not what you had in mind then please say :-)